hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/upgrade/TestDatanodeUpgradeToScmHA.java

```diff
@@ -103,6 +104,8 @@ public TestDatanodeUpgradeToScmHA(boolean scmHAAlreadyEnabled) {
     this.scmHAAlreadyEnabled = scmHAAlreadyEnabled;
     conf = new OzoneConfiguration();
     conf.setBoolean(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY, scmHAAlreadyEnabled);
+    // DATANODE_SCHEMA_V3 has higher feature version than SCM_HA
+    conf.setBoolean(DatanodeConfiguration.CONTAINER_SCHEMA_V3_ENABLED, false);
```

Comment:
What was the error seen here that led to this line being added? This test should not need to be updated every time we add a new layout feature.

Reply:
This still sounds like a bug though right? It is valid for a user to have ozone 1.1.0 (first layout version) and upgrade to Ozone 1.3.0 (schema v3) and turn on the schema v3 config when they do the upgrade. Schema v3 should still be disabled until the DN is finalized. When the DN finalizes, the rocksDB should be created and then schema v3 should work for all new containers.

Reply:
@errose28, the flow you described is correct and works.  The key point is rocksDB is created during upgrade from 1.1.0 to 1.3.0 (1.2.1 in the middle) when Schema V3 finalization action is executed. 
But in this test, it only upgrades like from 1.0.0 to 1.1.0,  only the SCM-HA related finalization action is executed, using cluster ID instead of scm ID as the data root directory.  Schema V3 finalization action will not executed, there is no change to create rocksDB for every volume.  In this case, if Schema V3 is enabled, then you can image what will happen.  You understand the difference now?

Reply:
@errose28 , thanks for the detail explain.  I now aware that all Finalization actions will be executed.  I reverted the changes in this test class with new commit.